ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/OptimizeService.java

```diff
@@ -146,12 +151,12 @@ public List<TableIdentifier> refreshAndListTables() {
 
     addNewTables(toAddTables);
     clearRemovedTables(toRemoveTables);
+    ArrayList<TableIdentifier> validTables = new ArrayList<>(cachedTables.keySet());
     if (!toAddTables.isEmpty() || !toRemoveTables.isEmpty()) {
-      internalCheckOptimizeChecker(
-          this.optimizeStatusCheckInterval == null ? DEFAULT_CHECK_INTERVAL : this.optimizeStatusCheckInterval);
+      internalCheckOptimizeChecker(validTables);
```

Comment:
Why should we schedule the check task here?

Reply:
To add the new tables/clear the removed tables as soon as possible

Reply:
I find `internalCheckOptimizeChecker` will be called twice, I will try to fix it.

Reply:
That's true, `internalCheckOptimizeChecker` will be called at least twice, if always have new tables are added while refresh tables, it will be called again and again.

Reply:
removed internalCheckOptimizeChecker from refreshAndListTables